envoy-control-services/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/ServiceInstances.kt

```diff
@@ -20,4 +20,11 @@ data class ServiceInstances(
                 .filter { it.address.isNotBlank() }
                 .toSet())
         } else this
+
+    fun withoutInvalidPortInstances(): ServiceInstances =
+        if (instances.any { it.port == 0 }) {
```

Comment:
it can be replaced, this check is unnecessary  
``` 
retrun copy(instances = instances.asSequence()
                .filter { it.port > 0 }
                .toSet())
```

Reply:
difference there is that i will always return a copy instead of already created object

Reply:
I think checking before copying is helpful here, because we do not want to make copy of all elements, if there is no service to filter out. (BTW. what will be the probability of this?)